doc/LaunchingVitess.md

```diff
@@ -0,0 +1,832 @@
+
+##### Transactions
+
+Transactions is a histogram variable that tracks transactions. The categories are "Completed" and “Aborted”.
+
+##### Waits
+
+Waits is a histogram variable that tracks various waits in the system. Right now, the only category is "Consolidations". A consolidation happens when one query waits for the results of an identical query already executing, thereby saving the database from performing duplicate work.
+
+This variable used to report connection pool waits, but a refactor moved those variables out into the pool related vars.
+
+##### Errors
+
+```
+
+  "Errors": {
+
+    "Deadlock": 0,
+
+    "Fail": 1,
+
+    "NotInTx": 0,
+
+    "TxPoolFull": 0
+
+  },
+
+```
+
+Errors are reported under different categories. It’s beneficial to track each category separately as it will be more helpful for troubleshooting. Right now, there are four categories. The category list may vary as Vitess evolves.
+
+Plotting errors/query can sometimes be useful for troubleshooting.
+
+VTTablet also exports an InfoErrors variable that tracks inconsequential errors that don’t signify any kind of problem with the system. For example, a dup key on insert is considered normal because apps tend to use that error to instead update an existing row. So, no monitoring is needed for that variable.
+
+##### InternalErrors
+
+```
+
+  "InternalErrors": {
+
+    "HungQuery": 0,
+
+    "Invalidation": 0,
+
+    "MemcacheStats": 0,
+
+    "Mismatch": 0,
+
+    "Panic": 0,
+
+    "Schema": 0,
+
+    "StrayTransactions": 0,
+
+    "Task": 0
+
+  },
+
+```
+
+An internal error is an unexpected situation in code that may possibly point to a bug. Such errors may not cause outages, but even a single error needs be escalated for root cause analysis.
+
+##### Kills
+
+```
+
+  "Kills": {
+
+    "Queries": 2,
+
+    "Transactions": 0
+
+  },
+
+```
+
+Kills reports the queries and transactions killed by VTTablet due to timeout. It’s a very important variable to look at during outages.
+
+##### TransactionPool*
+
+There are a few variables with the above prefix:
+
+```
+
+  "TransactionPoolAvailable": 300,
+
+  "TransactionPoolCapacity": 300,
+
+  "TransactionPoolIdleTimeout": 600000000000,
+
+  "TransactionPoolMaxCap": 300,
+
+  "TransactionPoolTimeout": 30000000000,
+
+  "TransactionPoolWaitCount": 0,
+
+  "TransactionPoolWaitTime": 0,
+
+```
+
+* WaitCount will give you how often the transaction pool gets full that causes new transactions to wait.
+
+* WaitTime/WaitCount will tell you the average wait time.
+
+* Available is a gauge that tells you the number of available connections in the pool in real-time. Capacity-Available is the number of connections in use. Note that this number could be misleading if the traffic is spiky.
+
+##### Other *Pool variables
```

Comment:
Oh that's fine, it just messes up the formatting I think. You have to add a
\ in front, like for underscore...

On Fri, Jun 3, 2016 at 10:12 AM, sougou <notifications@github.com> wrote:

> In doc/LaunchingVitess.md
> <https://github.com/youtube/vitess/pull/1757#discussion_r65742775>:
>
> > +
> > +  "TransactionPoolTimeout": 30000000000,
> > +
> > +  "TransactionPoolWaitCount": 0,
> > +
> > +  "TransactionPoolWaitTime": 0,
> > +
> > +```
> > +
> > +* WaitCount will give you how often the transaction pool gets full that causes new transactions to wait.
> > +
> > +* WaitTime/WaitCount will tell you the average wait time.
> > +
> > +* Available is a gauge that tells you the number of available connections in the pool in real-time. Capacity-Available is the number of connections in use. Note that this number could be misleading if the traffic is spiky.
> > +
> > +##### Other *Pool variables
>
> It's supposed to be regexp '*' for TransactionPool, etc. I can see why
> it's confusing. I'll just drop the star.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/youtube/vitess/pull/1757/files/749c4812810a0009fd2f67a67d30286513e41f92#r65742775>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEaVMtOF5Yafk_4KsKT_uhOXizDviBtUks5qIGBngaJpZM4ItuDf>
> .
>